src/library/scala/ClassValueCompat.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala
+
+import scala.util.Try
+
+private[scala] abstract class ClassValueCompat[T] {
```

Comment:
Sorry to bring this to discussion again, but do we really want to add this class to the `scala` package? Even if it's package private it will show up in some places way too prominently. For example when editing a Java file in a mixed project, `scala.ClassValueCompat` will be a public class.

I'd rather keep it in `scala.runtime`.

Reply:
agree; I argued against `scala.compat`, but I think `scala` is an even worse choice

Reply:
Issues mentioned below and here applied to the latest commit